plugins/rssurlrewrite/init.js

```diff
@@ -1,42 +1,40 @@
 plugin.loadMainCSS();
-if(browser.isIE && browser.versionMajor < 8)
+if (browser.isIE && browser.versionMajor < 8)
 	plugin.loadCSS("ie");
 plugin.loadLang();
 
-theWebUI.curRule = null;
-theWebUI.maxRuleNo = 0;
+plugin.curRule = null;
+plugin.maxRuleNo = 0;
+plugin.rules = [];
 
-theWebUI.showRules = function()
-{
-	theWebUI.request("?action=getrules",[theWebUI.loadRules, this]);
+function makeRuleListItem(rule, index) {
+	return $("<li>").append(
+		$("<input>")
+			.attr({type:"checkbox", id:`_re${index}`})
+			.prop("checked", rule.enabled),
+		$("<input>")
+			.attr({type:"text", id:`_rn${index}`})
+			.addClass("TextboxNormal")
+			.on("focus", selectRule)
+			.val(rule.name),
+	)
 }
 
-theWebUI.storeRuleParams = function()
-{
-	var no = 0;
-	if(this.curRule)
-	{
-		no = parseInt(this.curRule.id.substr(3));
-		this.rules[no].pattern = $('#RLS_pattern').val();
-		this.rules[no].replacement = $('#RLS_replacement').val();
-		this.rules[no].hrefAsSrc = $('#RLS_src').val();
-		this.rules[no].hrefAsDest = $('#RLS_dst').val();
-		this.rules[no].rssHash = $('#RLS_rss').val();
-	}
-	return(no);
-}
-
-theWebUI.selectRule = function( el )
-{
-	if(this.curRule!=el)
-	{
-		if(this.curRule)
-			this.curRule.className = 'TextboxNormal';
-		this.storeRuleParams();
-		this.curRule = el;
-		this.curRule.className = 'TextboxFocus';
-		var no = parseInt(this.curRule.id.substr(3));
-		var rle = this.rules[no];
+/**
+ * Actions to be performed when a RSS rule is selected.
+ * This function is going to be used as an event handler
+ * on rule's textbox focus events, so the `this` keyword
+ * refers to the event target, i.e. the focused rule item.
+ */
+function selectRule() {
```

Comment:
This function is only going to be used as an event handler for focusing on a rule list item, so a local function will work fine.